### DISCUSSION TITLE
How is EditorWidget width not an issue?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6198] [General]
Instead of posting this as an issue, I'll ask how it is *not* considered an issue already and if there is no satisfactory answer, I will post it as one. Thus:

`<$edit-text tiddler=HelloThere/>` or `<$edit-text tiddler=HelloThere size=100000 />` both show up as

![image](https://user-images.githubusercontent.com/4534056/141658460-f936b2fb-9b5a-4008-9571-d492492ddb73.png)

1. Why is the editor not 100% width (just like the editor in edit mode?)
2. Why *does* the editor have that arbitrary width? 
3. Why doesn't `size` do anything?
4. What is a user supposed to do if he doesn't know CSS to define some class to it?

If this were an issue request, I'd propose that 

1. default width is 100%
2. there's some parameter to set width for the textareas 
 
According to docs, the `size` property refers to "The size of the input field (in characters)". It is unclear if this means "total number of characters" in the editor or if it means "width" because the property can only be used in *single line* editors where it practically only affects the width. I have never had use of "total number of characters" so I think a property for width does make sense, whether this is really what `size` means or whether there is need to introduce a new `width` parameter.

**Comment by pmario** [comment]
There are a lot of parameters. I'd try to set them. https://tiddlywiki.com/#EditTextWidget

`size` .. probably is for tag=input and not tag=textarea 

```
<$edit-text tiddler=HelloThere class="tc-edit-texteditor tc-edit-texteditor-body" autoHeight=no />
```

Works just fine. BUT the autoHeight activates the height setting from the editmode. ... So there's probably a CSS specificity problem.

**Reply by twMat** [reply]
Thanks @pmario 

>`size` .. probably is for tag=input and not tag=textarea

OK, so should be stated in the docs. I'll review it.  

>`<$edit-text tiddler=HelloThere class="tc-edit-texteditor tc-edit-texteditor-body" autoHeight=no />`

Thanks, AFAICT it seems to be enough with:

`<$edit-text tiddler=HelloThere class="tc-edit-texteditor" />`

So, this example should also be in the docs. It also solves the autoHeight in your example. 

**But why should 100% width *not* be the default? What justifies the current magic width?**

**Reply by Jermolene** [reply]
> But why should 100% width not be the default? What justifies the current magic width?

A plain, unadorned edit-text widget with tag set to "textarea" is rendered as a plain HTML textarea element with no special styling. If we applied a default class or set of styles then we'd need to provide a way to disable them because they wouldn't be appropriate in all cases.

**Reply by twMat** [reply]
>[...] rendered as a plain HTML textarea element

Fair enuff. I still wonder what the magic default width *does* come from but I guess that's a question for the W3C or something.

**Comment by pmario** [comment]
>It also solves the autoHeight in your example.

I think `autoHeight` should be set to "yes" =auto by default

**Reply by twMat** [reply]
>I think autoHeight should be set to "yes" =auto by default

autoHeight=yes *is* default, according to the docs. If that is what you refer to.
